### PR TITLE
add pod labels to spire server values

### PR DESCRIPTION
**Pull request by mcrors**
The reason for this change is to be able to integrate spire with Azure Workload Idenity. To enable this, a label of `azure.workload.identity/use: "true"` needs to be added to the pods metadata.

https://learn.microsoft.com/en-us/azure/aks/learn/tutorial-kubernetes-workload-identity#deploy-the-workload

### CODE REVIEW

File: charts/spire/charts/spire-server/templates/statefulset.yaml [hunk 1]
```diff
@@ -43,6 +43,9 @@ spec:
         component: server
         release: {{ .Release.Name }}
         release-namespace: {{ .Release.Namespace }}
+        {{ with .Values.podLabels }}
```
Comment (marcofranssen) [inline charts/spire/charts/spire-server/templates/statefulset.yaml:46]:
```suggestion
        {{- with .Values.podLabels }}
```

**Review by marcofranssen — CHANGES_REQUESTED** [review]
To remove the whiteline from the rendered yaml, please add the suggestion please.

File: charts/spire/charts/spire-server/templates/statefulset.yaml (hunk 1 above)
Comment (mcrors) [reply charts/spire/charts/spire-server/templates/statefulset.yaml]: Cheers.

**Review by kfox1111 — APPROVED** [review]
LGTM. Thanks! :)

**Pull request merged**